app/models/product_share.rb

```diff
@@ -0,0 +1,29 @@
+class ProductShare < ApplicationRecord
+  include PowerTypes::Observable
+  belongs_to :product
+  enum platform: { email: 0, whatsapp: 1 }
+
+  validates :to_email, presence: true, format: Devise::email_regexp
```

Comment:
qué va a pasar cuando se comparta por whatsapp? se va a caer o no?

Reply:
Tienes toda la razón, al compartir por WSP el email queda vacío, y tampoco hay otra info para "rellenar" (el equipo decidió no pedirle el numero al usuario), por lo cual la validación se va a caer.

La solución mas fácil por el momento seria **sacar la validación**. No estoy seguro si esto sea lo mejor. Si te puedo decir que en el siguiente PR que tengo preparado hago una validación del mail por front-end.

Reply:
buena! igual te dejo esto por si te sirve: https://guides.rubyonrails.org/active_record_validations.html#using-a-symbol-with-if-and-unless 👍

Reply:
buena DDD:, eso me gusta mas! Pensé que a lo mejor se podía, pero no sabia que uno podía hacer una validación condicional tan fácil. Gracias